### PR TITLE
util: remove `Error{,SkipFrames}`

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

**Comment by mberhault** [comment]
this is entirely overkill. There is no downside to using util.Errorf in tests and makes things a lot simpler to find. Additionally, a lot of internal errors should keep Errorf, even when presented to the user, this will make debugging much quicker.

**Comment by tamird** [comment]
I think there is one case where util.Errorf changed to errors.New. The rest
of the changes are from util.Errorf to util.Error.

**Comment by tamird** [comment]
Or from fmt.Errorf to errors.New

**Comment by bdarnell** [comment]
I didn't realize we had `util.Error`, but now that I see it I think it's a bit silly to have both. If we're going to make sweeping changes in this area we should just replace all uses of `util.Error` with `util.Errorf` (but to be clear the status quo is perfectly acceptable). 

The advantage of `Error` over `Errorf` is that you can pass something other than a literal string for the first argument, and you don't need to add "%s" when there are more than one argument, but in no case (out of the 29 current uses of `util.Error`, with the sole exception of its own test case) do we take advantage of that (nor do I think we should).

**Comment by tamird** [comment]
done

File: util/error.go [hunk 1]
```diff
@@ -48,12 +48,6 @@ func ErrorfSkipFrames(skip int, format string, a ...interface{}) error {
...
 
-// Error is a passthrough to fmt.Error, with an additional prefix
-// containing the filename and line number.
-func Error(a ...interface{}) error {
-	return ErrorSkipFrames(1, a...)
-}
-
 // ErrorSkipFrames allows the skip count for stack frames to be
 // specified. See the comments for ErrorfSkip.
 func ErrorSkipFrames(skip int, a ...interface{}) error {
```
Comment (bdarnell) [inline util/error.go]: Also remove ErrorSkipFrames (which has no uses besides its own test).

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**